bundles/org.openhab.binding.magentatv/README.md

```diff
@@ -0,0 +1,259 @@
+| FORWARD  | Forward                                        |
+| LASTCH   | Last channel                                   |
+| NEXTCH   | Next channel                                   |
+| LASTCHAP | Last chapter                                   |
+| NEXTCHAP | Next chapter                                   |
+| PAIR     | Re-pair with the receiver                      |
+
+In addition you could send any key code in the 0xHHHH format., refer to
+[Key Codes for Magenta/Huawei Media Receiver](http://support.huawei.com/hedex/pages/DOC1100366313CEH0713H/01/DOC1100366313CEH0713H/01/resources/dsv_hdx_idp/DSV/en/en-us_topic_0094619112.html)
+
+
+## Full Configuraton Example
+
+### magentatv.things
+
+```
+Thing magentatv:receiver:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx "MagentaTV" [
+udn="xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
+ipAddress="xxx.xxx.xxx.xxx", 
+accountName="xxxxxx.xxxx@t-online.de",
+accountPassword="xxxxxxxxxx"
+]
+```
+
+### magentatv.items
+
+```
+# MagentaTV Control
+Switch MagentaTV_Power        "Power"        {channel="magentatv:receiver:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx:control#power"}
+Number MagentaTV_Channel      "Channel"      {channel="magentatv:receiver:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx:status#channel"}
+String MagentaTV_Key          "Key"          {channel="magentatv:receiver:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx:control#key"}
+
+# MagentaTV Program Information
+String MagentaTV_ProgTitle   "Program Title" {channel="magentatv:receiver:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx:program#title"}
+String MagentaTV_ProgDescr   "Description"   {channel="magentatv:receiver:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx:program#text"}
+String MagentaTV_ProgStart   "Start Time"    {channel="magentatv:receiver:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx:program#tart"}
+String MagentaTV_ProgDur     "Duration"      {channel="magentatv:receiver:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx:program#duration"}
+String MagentaTV_ProgPos     "Position"      {channel="magentatv:receiver:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx:program#position"}
+
+# MagentaTV Play Status
+Number MagentaTV_ChCode    "Channel Code"    {channel="magentatv:receiver:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx:status#channelCode"}
+String MagentaTV_PlayMode  "Play Mode"       {channel="magentatv:receiver:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx:status#playMode"}
+String MagentaTV_RunStatus "Run Status"      {channel="magentatv:receiver:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx:status#runStatus"}
+```
+
+### sitemap
+
+```
+please contribute an example
+```
+
+### magentatv.rules
+
+Due to the fact the POWER is a toggle button and the binding cannot detect the current status, which could lead into the situation that you want to power on the receiver as part of a scene, but due to the fact that it is already ON you switch it off.
+
+Beginning with models 401/201 and new the binding is able to detect the Power-OFF condition, which can be used in a rule to improve this situation
```

Comment:
The binding is not able to read the status. The only improvement I found was to catch the UPnP event on power-off + POWER is a toggle button, you can't switch ON or OFF you can only toggle the current status. In combination with the fact that the status is unknown you depend on the current status. I also had a version, which tries to keep the status in the binding, but practically this doesn't work out well.

Reply:
That still does not explain what the rule is meant to do other than what could be implemented in the binding already.

Reply:
If the binding is not transparent and the status is not in sync you are never able to send the POWER key, e.g.
- OH starts, receiver is off
- OH sends an POWER, receiver goes ON
- now you press the power button on the remote = receiver OFF vs. binding state ON
- you run a scene, which expects the receiver to be ON
- the binding doesn't send ON, because it's status is still ON

With the MR401 the beding detects the manual POWER OFF, but not with the MR400
In the current implementation you could bring both in sync just by sending POWER manually, which is send to the receiver, because the binding is transparent.

Reply:
So what you are saying is that although you can identify the power state on an MR401, this state is still unreliable, so that the user needs a way to bring the real receiver state and the item state back in sync (and this is only possible if the logic is not in the binding, but only in external rules)?

Reply:
ok, we have

MR 400/MR200
- does not send SSDN messages when shutting down
- responds to ping when network interface is active, but takes about 45s on start-up
- no way to detect display status or state of the power button

MR 401/201
- sends SSDN message on shutdown, but nothing on startup
- does not respond to ping - so, this can't be used to detect power status
- no way to detect display status or state of the power button
- answers to http requests event when in state OFF (not in sleep mode) -> can't use dummy request to see if the device is awake
- sometimes does an UPnP advertisement (which is intercepted by the binding)
- sends program events when a channel is switched, but not initially when the MR is turned ON

+ 
- POWER is a toggle button
- if you want to start your scene while the receiver is ON you'll switch it OFF

I think it's a bad idea to build a logic into the binding, which relies on SSDN shutdown message, but is not able to detect that the receiver is ON, we need to differ between
- MR is in deep sleep / not connected to power
- MR is in standby mode, responds to network requests, but doesn't report the POWER status
- MR is active and playing a TV channel
- MR is active and playing streaming content

The SSDN message helps to get the POWER status in sync, but I didn't find a way to see that the MR is not only responding to network request, but is aware and playing content. In my UI I have a button sending POWER and one starting a scene (switching TV and MR on, selecting HDMI etc.). The scene uses the rule and usually I need only that button. The rule checks (the binding's) POWER state and sends ON if the MR is currently off. If for any reason the OH status and MR status get out of sync  I have the "raw" POWER button to fix that.

I spend several hours to find any indicator, which helps to overcome this issue, but without success. Maybe you have the magic idea. Any by the way: Nobody is obligated to use the external rule, but an incomplete logic in the binding could screw up the user experience with no way to change it.